app/src/main/java/de/danoeh/antennapod/activity/BugReportActivity.java

```diff
@@ -67,42 +72,60 @@ protected void onCreate(Bundle savedInstanceState) {
+                    File filename = new File(UserPreferences.getDataFolder(null), "full-logs.txt");
+                    filename.createNewFile();
+                    String cmd = "logcat -d -f " + filename.getAbsolutePath();
+                    Runtime.getRuntime().exec(cmd);
+                    //share file
+                    try {
+                        Intent i = new Intent(Intent.ACTION_SEND);
+                        i.setType("text/*");
+                        String authString = getString(de.danoeh.antennapod.core.R.string.provider_authority);
+                        Uri fileUri = FileProvider.getUriForFile(this, authString, filename);
+                        i.putExtra(Intent.EXTRA_STREAM, fileUri);
+                        i.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+                        if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.KITKAT) {
+                            PackageManager pm = getPackageManager();
+                            List<ResolveInfo> resInfos = pm.queryIntentActivities(i, PackageManager.MATCH_DEFAULT_ONLY);
+                            for (ResolveInfo resolveInfo : resInfos) {
+                                String packageName = resolveInfo.activityInfo.packageName;
+                                grantUriPermission(packageName, fileUri, Intent.FLAG_GRANT_READ_URI_PERMISSION);
+                            }
                         }
+                        String chooserTitle = getString(de.danoeh.antennapod.core.R.string.share_file_label);
+                        startActivity(Intent.createChooser(i, chooserTitle));
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                        int strResId = R.string.log_file_share_exception;
+                        Snackbar.make(findViewById(android.R.id.content), strResId, Snackbar.LENGTH_LONG)
+                                .show();
                     }
-                    String chooserTitle =  getString(de.danoeh.antennapod.core.R.string.share_file_label);
-                    startActivity(Intent.createChooser(i, chooserTitle));
-                } catch (Exception e) {
+                } catch (IOException e) {
                     e.printStackTrace();
-                    int strResId = R.string.log_file_share_exception;
-                    Snackbar.make(findViewById(android.R.id.content), strResId, Snackbar.LENGTH_LONG)
-                            .show();
+                    Snackbar.make(findViewById(android.R.id.content), e.getMessage(), Snackbar.LENGTH_LONG).show();
                 }
-            } catch (IOException e) {
-                e.printStackTrace();
-                Snackbar.make(findViewById(android.R.id.content), e.getMessage(), Snackbar.LENGTH_LONG).show();
-            }
-        });
+                dialog.dismiss();
+            });
+            alertBuilder.setNegativeButton("Cancel", (dialog, which) -> dialog.cancel());
+            alertBuilder.create();
```

Comment:
Calling create is not needed if you show it afterwards. You can just delete this line.